test/api/compilation.js

```diff
@@ -0,0 +1,270 @@
+'use strict';
+
+const expect = chai.expect;
+const assert = chai.assert;
+
+import * as utils from '../utils.js';
+
+describe('test Compilation', function() {
+  const nn = navigator.ml.getNeuralNetworkContext();
+  const builder = nn.createModelBuilder();
+  const desc = {type: 'float32', dimensions: [2, 2]};
+  const a = builder.input('a', desc);
+  const b = builder.input('b', desc);
+  const c = builder.matmul(a, b);
+  const d = builder.constant(
+      {type: 'float32', dimensions: [2, 2]}, new Float32Array(4).fill(1));
+  const e = builder.add(c, d);
+  const bufferA = new Float32Array(4).fill(1);
+  const bufferB = new Float32Array(4).fill(1);
+  const expectedC = [2, 2, 2, 2];
+  const expectedE = [3, 3, 3, 3];
+
+  it('Compilation should have compute method', async () => {
+    const model = builder.createModel({c});
```

Comment:
this is bit confusing, the builder creates models, but the topology is created before the model is created, and how made the topology immutable for the model?

Reply:
The model is immutable after `createModel`. As discussed with @pyu10055 , it needs to add details of the `createModel` method into the spec. Created https://github.com/webmachinelearning/webnn/issues/107 to follow up.

Reply:
I added a new commit https://github.com/webmachinelearning/webnn-polyfill/pull/23/commits/e0ba924435282afff0bc3984218855a3bca947d1. It includes a test case for immutable model and fixes a bug that the constant operand is not immutable. Please take another look.